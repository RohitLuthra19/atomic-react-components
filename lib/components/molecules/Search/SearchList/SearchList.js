// @flow
import React, { Node } from 'react';
import styled from 'styled-components';

import type { Props } from './types';
import styles from './SearchList.style';
import List from '../../List';

const SearchList = ({ dataLists, renderList }: Props): Node => {
  return (
    <List ListType="ul">
      {dataLists.map(val => {
        return (
          <li className="search-list" key={val.id}>
            {renderList(val.name)}
          </li>
        );
      })}
    </List>
  );
};

SearchList.defaultProps = {
  className: '',
  renderList: () => {},
  dataLists: ['test', 'test2'],
};

export default styled(SearchList)`
  ${styles};
`;

export { SearchList as SearchListVanilla };
