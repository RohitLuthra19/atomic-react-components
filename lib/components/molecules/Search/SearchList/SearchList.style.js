import { css } from 'styled-components';

export default css`
  .search-list {
    background: #fff;
    display: flex;
    border: 1px solid #dfe1e5;
    box-shadow: none;
    width: 638px;
    border-radius: 24px;
    z-index: 3;
    height: 44px;
    margin: 0 auto;
    border-radius: 5px;
    margin-top: 10px;
    line-height: 40px;
    padding-left: 12px;
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
