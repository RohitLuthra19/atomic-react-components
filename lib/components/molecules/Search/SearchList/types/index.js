// @flow
export type Props = {
  className?: string,
  dataLists?: object,
  renderList: func,
};
