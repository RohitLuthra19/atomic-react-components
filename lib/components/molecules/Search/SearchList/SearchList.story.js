import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './SearchList.mock';

// Import Styled Component to showcase variations
import SearchList, { SearchListVanilla } from './SearchList';

const ListItem = ({ children }) => children;

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module)
  .addParameters({ jest: ['Search', 'SearchVanilla'] })
  .add('Search List Knobs', () => (
    <SearchListVanilla
      {...defaultConfig}
      className={`hide-default-sample ${defaultConfig.className}`}
    />
  ))
  .add('Search List', () => (
    <SearchList {...defaultConfig} renderList={item => <ListItem key={item}>{item}</ListItem>} />
  ));
