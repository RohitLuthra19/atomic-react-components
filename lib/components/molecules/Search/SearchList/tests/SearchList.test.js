import { shallow } from 'enzyme';

import SearchList from '../SearchList';

describe('<SearchList />', () => {
  let SearchListComponent = '';
  beforeEach(() => {
    SearchListComponent = shallow(<SearchList>Test</SearchList>);
  });

  test('should render correctly', () => {
    expect(SearchListComponent).toMatchSnapshot();
  });
});
