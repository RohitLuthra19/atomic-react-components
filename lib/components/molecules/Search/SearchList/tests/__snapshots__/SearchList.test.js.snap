// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchList /> should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SearchList-sc-1z8pyj-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";",
          ";",
        ],
      },
      "displayName": "SearchList",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SearchList-sc-1z8pyj-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Test
</StyledComponent>
`;
